### PR TITLE
token auth with feature flag

### PR DESCRIPTION
**Pull request by Connoropolous**
feature flag is 
FEATURE_FLAG_GENERATE_TOKEN
intention would be to initially set this to 'test'

demo: 
https://youtu.be/jf7BYkECVtA

### CODE REVIEW

**Comment by Connoropolous** [comment]
goes with https://github.com/Hylozoic/hylo-node/pull/190

File: src/containers/user/UserSettings.js [hunk 1]
```diff
@@ -193,10 +196,20 @@ export default class UserSettings extends React.Component {
     this.delayedUpdate(field, value)
   }
 
+  generateToken = () => {
+    const { dispatch } = this.props
+    this.setState({tokenPending: true})
+    dispatch(generateUserToken())
+    .then(action => action.error ? this.setState({tokenError: true, tokenPending: false}) : this.setState({receivedToken: action.payload.accessToken, tokenPending: false}))
```
Comment (levity) [inline src/containers/user/UserSettings.js]: let's wrap this long line as:
```js
action.error
  ? ...
  : ...
```

File: src/containers/user/UserSettings.js [hunk 2]
```diff
@@ -391,6 +404,29 @@ export default class UserSettings extends React.Component {
           </Item>
         </Section>}
       </div>}
+      
+      {hasFeature(currentUser, GENERATE_TOKEN) && <div>
+        <SectionLabel name='developer' label='API Access' {...{dispatch, expand}}/>
+        {expand.developer && <Section className='apiAccess'>
+          <Item>
+            <div className='full-column'>
+              { !hasToken && !receivedToken && <p>Generate a token with which you can access the Hylo API</p>}
```
Comment (levity) [inline src/containers/user/UserSettings.js]: no space after the `{` please

File: src/reducers/currentUser.js [hunk 3]
```diff
@@ -99,6 +101,16 @@ export default function (state = null, action) {
           viewedTooltips: {}
         }
       }
+    case GENERATE_USER_TOKEN:
+      return {
+        ...state,
+        linkedAccounts: state.linkedAccounts.concat([{ provider_key: 'token'}])
```
Comment (levity) [inline src/reducers/currentUser.js]: no space after `{` please

**Review by levity — COMMENTED** [review]
Make the suggested changes and we'll be all set

**Comment by Connoropolous** [comment]
👍

**Pull request merged**